### PR TITLE
added user admin role and updated researcher and site admin permissions

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- User Admin role added as per DEX-191
- Updated Researcher and site admin permissions to reflect  the latest role definition 

---

**Pull Request Checklist**
- [x] Ensure that the PR is properly formatted
  - Example: All lint checks are passing
- [x] Ensure that the PR is properly rebased
  - Example: The PR is rebased on `develop` (commit: `<insert develop commit hash>`)
- [x] Ensure that the PR uses a consolidated database migration
  - Example: One database migration is proposed (revision `<insert develop revision>` -> `<insert new revision>`)
- [x] Ensure that the PR is properly tested
  - Example: All automated tests are passing
- [x] Ensure that the PR is properly sanitized
  - Example: No sensitive data or large content was added to this PR
- [x] Ensure that the PR is properly reviewed

### CODE REVIEW

File: app/modules/users/permissions/rules.py [hunk 1]
```diff
@@ -130,19 +130,17 @@ def _can_user_perform_action(self, user):
...
 
         elif self._action is AccessOperation.WRITE:
             if self._is_module(HoustonConfig):
                 has_permission = user.is_admin
-            elif self._is_module((Submission, User)):
-                # Any users can submit and write (create) a user
-                has_permission = True
-            elif self._is_module(Encounter):
+            # Any users can write (create) a user, submission and Encounter, TODO, decide on Submission
+            elif self._is_module((Submission, User, Encounter)):
```
Comment (colinwkingen) [inline app/modules/users/permissions/rules.py:143]: Good catch here, I think that may have been some merge cruft I was responsible for.

File: tests/modules/users/test_permissions.py [hunk 2]
```diff
@@ -349,7 +349,28 @@ def test_ModuleAccessPermission_admin_user(admin_user_login):
...
+
+def test_ModuleAccessPermission_user_admin_user(user_admin_user_login):
+    # pylint: disable=unused-argument
+    from app.modules.users.models import User
+    from app.modules.submissions.models import Submission
+
+    # user Admins not especially priviliged
+    validate_cannot_read_module(NotARealClass)
+    validate_cannot_write_module(NotARealClass)
+    validate_cannot_delete_module(NotARealClass)
```
Comment (colinwkingen) [inline tests/modules/users/test_permissions.py:371]: This `NotARealClass` is just to ensure checks have the ability to fail, yes?

**Review by colinwkingen — DISMISSED** [review]
@hwindsor We already talked about the name change for the role, I'm ready to merge this in when that happens.

**Comment by colinwkingen** [comment]
This should have been a 'Comment' review not an Approve, but I'll leave it as-is and merge once the change from user admin to user manager is complete.

File: tests/modules/users/test_permissions.py (hunk 2 above)
Comment (hwindsor) [reply tests/modules/users/test_permissions.py:371]: it's a check that the user doesn't have access to absolutely anything, like admin used to have and staff still do

**Review by colinwkingen — APPROVED** [review]

**Pull request merged**